src/sunadaptcontroller/soderlind/sunadaptcontroller_soderlind.c

```diff
@@ -277,6 +277,65 @@ SUNErrCode SUNAdaptController_SetParams_ImpGus(SUNAdaptController C,
   return SUN_SUCCESS;
 }
 
+/* -----------------------------------------------------------------
+ * Function to create an H_{0}211 controller (subset of Soderlind)
+ */
+
+SUNAdaptController SUNAdaptController_H0211(SUNContext sunctx)
+{
+  SUNFunctionBegin(sunctx);
+
+  SUNAdaptController C = SUNAdaptController_Soderlind(sunctx);
+  SUNCheckLastErrNull();
+
+  SUNCheckCallNull(
+    SUNAdaptController_SetParams_Soderlind(C, SUN_RCONST(0.5), SUN_RCONST(0.5),
+                                           SUN_RCONST(0.0), -SUN_RCONST(0.5),
+                                           SUN_RCONST(0.0)));
+
+  return (C);
+}
+
+/* -----------------------------------------------------------------
+  * Function to create an H211 controller (subset of Soderlind)
+  */
+
+SUNAdaptController SUNAdaptController_H211(SUNContext sunctx)
+{
+  SUNFunctionBegin(sunctx);
+
+  SUNAdaptController C = SUNAdaptController_Soderlind(sunctx);
+  SUNCheckLastErrNull();
+
+  SUNCheckCallNull(
+    SUNAdaptController_SetParams_Soderlind(C, SUN_RCONST(0.25),
+                                           SUN_RCONST(0.25), SUN_RCONST(0.0),
+                                           -SUN_RCONST(0.25), SUN_RCONST(0.0)));
+
+  return (C);
+}
+
+/* -----------------------------------------------------------------
+  * Function to create an H312 controller (subset of Soderlind)
+  */
+
+SUNAdaptController SUNAdaptController_H312(SUNContext sunctx)
```

Comment:
Should this be H0312 or H312?
```suggestion
SUNAdaptController SUNAdaptController_H0312(SUNContext sunctx)
```

Reply:
This is the H312b controller that uses the "recommended" value of b=8.  Thus I think that the name H312 makes the most sense, and we'll just need to remove other occurrences of H0312 (if any).

Reply:
Thanks for confirming. I think the only occurrences of H_{0}312 are in the FAQ entry discussing results with the various controllers. Should these have been H312?

Reply:
No, the FAQ also included H312.  The problem is that when I updated the FAQ I included both H312 and H_{0}312, but since our tests had only recommended H312 I ended up not implementing H_{0}312.  I've just pushed a fix to the FAQ to remove H_{0}312 since it is not included with SUNDIALS.